### PR TITLE
Enable using built-in libexpat on Linux

### PR DESCRIPTION
**Pull request by MaartenBent**
Required to make https://github.com/wxWidgets/wxWidgets/commit/f25a88a13dc51d9d857e10b4b264c0e817bb7a0f actually work

### CODE REVIEW

File: expat/lib/xmltok.c [hunk 1]
```diff
@@ -42,15 +42,13 @@
 #  include <stdbool.h>
 #endif
 
-#ifdef _WIN32
+#if defined(_WIN32)
```
Comment (vadz) [inline expat/lib/xmltok.c]: Does this change come from the upstream sources?

**Review by vadz — COMMENTED** [review]
No real comments, just a small remark: it should be "built-in" in all the commit messages.

File: expat/lib/xmltok.c (hunk 1 above)
Comment (MaartenBent) [reply expat/lib/xmltok.c]: No, I just replaced it so it is consistent with `#elif defined()`. Should I revert it?

Comment (MaartenBent) [reply expat/lib/xmltok.c]: I can change it to the following. Then there are only two extra lines compared to upstream.
```c++
#ifdef _WIN32
#  include "winconfig.h"
#else
#  ifdef HAVE_EXPAT_CONFIG_H
#    include <expat_config.h>
#  else
#    include "macconfig.h"
#  endif
#endif /* ndef _WIN32 */
```

Comment (vadz) [reply expat/lib/xmltok.c]: Let's do it like this, thanks!

**Comment by vadz** [comment]
Thanks!

Should I update submodules in the main repository or will you do it as part of another PR with more modifications to CMake files?

**Comment by MaartenBent** [comment]
Yes, please update the submodules in the main repository. Thanks.

**Pull request merged**